docs/get-started/getting-started-kubernetes.md

```diff
@@ -0,0 +1,256 @@
+how to install a Kubernetes Cluster with NVIDIA GPU support.
+
+## Docker Image Preparation
+
+On a client machine which has access to the Kubernetes Cluster:
+
+1. Download Spark from https://spark.apache.org/downloads.html
+   Supported versions of Spark are listed on the [download](../download.md) page.  Please note that only
+   scala version 2.12 is currently supported by the accelerator. 
+
+   Note that you can download these into a local directory and untar the Spark `.tar.gz` as a directory named `spark`.
+
+2. Download [RAPIDS Accelerator for Spark jars](getting-started-on-prem.md#download-the-rapids-jars), and the
+  [GPU discovery script](getting-started-on-prem.md#install-the-gpu-discovery-script).
+  
+   Put the 2 jars -- `rapids-4-spark_<version>.jar`, `cudf-<version>.jar`  and `getGpusResources.sh` in the same directory as `spark`.
+
+3. Download the sample [Dockerfile.cuda](Dockerfile.cuda) in the same directory as `spark`.
+
+   Basically the sample Dockerfile.cuda will copy the `spark` directory's several sub-directories into `/opt/spark/`, 
+   and copy above 2 jars and `getGpusResources.sh` into `/opt/sparkRapidsPlugin` inside the Docker image.
+   
+   So please look into the Dockerfile.cuda to make sure the file names are correct, and modify if needed.
+   
+   Currently the directory in the local machine should look as below:
+   ```shell 
+   $ ls
+   Dockerfile.cuda   cudf-<version>.jar   getGpusResources.sh   rapids-4-spark_<version>.jar   spark
+   ```
+
+4. Build the Docker image with proper repository name and tag and push to repository
+   ```shell 
+   export IMAGE_NAME=xxx/yyy:tag
+   docker build . -f Dockerfile.cuda -t $IMAGE_NAME
+   docker push $IMAGE_NAME
+   ```
+
+## Run Spark Applications in Kubernetes Cluster
+
+### spark-submit example pi job in `cluster` mode.
+
+This is a good way to test if RAPIDS plugin can be found.
+Because `ClassNotFoundException` is a common error if the Driver can not 
+find the jar for `com.nvidia.spark.SQLPlugin`:
+```
+Exception in thread "main" java.lang.ClassNotFoundException: com.nvidia.spark.SQLPlugin
+```
+
+Here is an example job: 
+
+```shell
+export SPARK_HOME=~/spark
+export IMAGE_NAME=xxx/yyy:tag
+export K8SMASTER=k8s://https://<k8s-apiserver-host>:<k8s-apiserver-port>
+export SPARK_NAMESPACE=default
+export SPARK_DRIVER_NAME=exampledriver
+
+$SPARK_HOME/bin/spark-submit \
+     --master $K8SMASTER \
+     --deploy-mode cluster  \
+     --name examplejob \
+     --class org.apache.spark.examples.SparkPi \
+     --conf spark.rapids.sql.concurrentGpuTasks=1 \
+     --conf spark.executor.instances=1 \
+     --conf spark.executor.resource.gpu.amount=1 \
+     --conf spark.executor.memory=4G \
+     --conf spark.executor.cores=1 \
+     --conf spark.task.cpus=1 \
+     --conf spark.task.resource.gpu.amount=1 \
+     --conf spark.rapids.memory.pinnedPool.size=2G \
```

Comment:
If we're going to ask for 2GB of pinned memory outside of the executor heap then I think we need to set `spark.kubernetes.memoryOverheadFactor` accordingly so Kubernetes expects the extra memory usage.